doc_src/terminal-compatibility.rst

```diff
@@ -0,0 +1,188 @@
+Terminal Compatibility
```

Comment:
everything but soft-wrapping is about control sequences here,
maybe we should call it `terminal-control-sequences` or `terminal-ctlseqs`.

I wonder what else we could put in there.
`fish_handle_reflow` comes to mind (that's one of the remaining places where we do user-agent-sniffing)
But if we switch to a different solution here, it will likely be based on a control sequence as well.
So I'll try the rename.

Reply:
actually we might want to document what kinds of key encodings we support (in addition to the kitty keyboard protocol).
Probably keeping the name I guess